extraction/extractor.py

```diff
@@ -77,16 +78,20 @@ def __extract_data_from_html_body(self, node):
         # Add to the final list if a valid content is extracted
         if _content:
             self.contents_list.add_content(_content)
+            return
 
-        if node and node.name and not _content:
+        if node and node.name:
             for child in node.contents:
                 self.__extract_data_from_html_body(child)
 
     def __validate_and_extract_content(self, node):
         """This function will extract valid content and return it"""
 
-        # This function will be updated in next iteration
-        _ = node
         _ = self.soup
+        # checks if content is image
+        content_extractor = image_extractor.ImageExtractor()
+        image_content = content_extractor.validate_and_extract(node)
+        if image_content:
+            return image_content
```

Comment:
Create a list of all IContentExtractors and loop around it.

Reply:
Updated!